src/providers/ipa/ipa_deskprofile_config.c

```diff
@@ -0,0 +1,158 @@
+/*
+    SSSD
+
+    Authors:
+        Fabiano Fidêncio <fidencio@redhat.com>
+
+    Copyright (C) 2017 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "util/util.h"
+#include "providers/ipa/ipa_deskprofile_private.h"
+#include "providers/ipa/ipa_deskprofile_config.h"
+#include "providers/ldap/sdap_async.h"
+
+struct ipa_deskprofile_config_state {
+    struct sysdb_attrs *config;
+};
+
+static void
+ipa_deskprofile_get_config_done(struct tevent_req *subreq);
+
+struct tevent_req *
+ipa_deskprofile_get_config_send(TALLOC_CTX *mem_ctx,
+                                struct tevent_context *ev,
+                                struct sdap_handle *sh,
+                                struct sdap_options *opts,
+                                const char *domain)
+{
+    struct tevent_req *req = NULL;
+    struct tevent_req *subreq;
+    struct ipa_deskprofile_rule_state *state;
+    char *ldap_basedn;
+    char *basedn;
+    char *rule_filter;
+    const char *attrs[] = { IPA_DESKPROFILE_PRIORITY, NULL };
+    errno_t ret;
+
+    req = tevent_req_create(mem_ctx, &state,
+                            struct ipa_deskprofile_config_state);
+    if (req == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "tevent_req_create() failed.\n");
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = domain_to_basedn(state, domain, &ldap_basedn);
+    if (ret != EOK) {
+        DEBUG(SSSDBG_OP_FAILURE, "domain_to_basedn() failed.\n");
+        goto done;
+    }
+
+    basedn = talloc_asprintf(state, IPA_DESKPROFILE_BASE_TMPL, ldap_basedn);
```

Comment:
Why does `ipa_get_id_options` allow setting the search base if we then use a hardcoded one?